core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1331,6 +1337,9 @@ object Parser extends ParserInstances {
         case Map(p, _) =>
           // we discard any allocations done by fn
           unmap(p)
+        case Select(p, _) =>
+          // we discard any allocations done by fn
+          unmap(p)
```

Comment:
I think you need to do Select(a, b) => Select(a, unmap(b))

Since in select, after we optionally parse the right in a void we could not do anything more, but we do need the result on the left even in a void.

Reply:
The second parameter has to be a function, so I had to immediately and trivially map the unmapped value.